vars.tf

```diff
@@ -4,14 +4,8 @@ variable "bus_name" {
 }
 
 variable "targets" {
-  type = map(string)
-
-  description = "Targets to route event to. Must specify an `arn = type`. `type` must be one of `lambda` or `bus`."
-
-  validation {
-    condition     = alltrue([for arn, type in var.targets : contains(["lambda", "bus"], type)])
-    error_message = "Invalid `type` given. Valid values are 'lambda' or 'bus'."
-  }
+  type        = set(string)
+  description = "Targets to route event to"
```

Comment:
How about using an object with optional attributes instead? This would enforce some constraints and avoid the messy string parsing.
```tf
type = object({
  lambda = optional(list(string))
  bus = optional(list(string))
})